File: app/src/main/java/oop2/ex01.java
package oop2;

public class ex01 {
    static class Employee {
        protected String name;

        Employee(String name) {
            this.name = name;
        }

        void work() {
            System.out.println(name + " works");
        }
    }

    static class Manager extends Employee {
        Manager(String name) {
            super(name);
        }

        @Override
        void work() {
            System.out.println(name + " manages team");
        }
    }

    public static void main(String[] args) {
        Employee manager = new Manager("Chris");
        manager.work();
    }
}

File: app-kotlin/src/main/kotlin/oop2/ex01.kt
package oop2

class ex01 {
    open class Employee(protected val name: String) {
        open fun work() {
            println("$name works")
        }
    }

    class Manager(name: String) : Employee(name) {
        override fun work() {
            println("$name manages team")
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val manager: Employee = Manager("Chris")
            manager.work()
        }
    }
}
